src/clusterfuzz/_internal/bot/tasks/utasks/uworker_io.py

```diff
@@ -219,334 +150,23 @@ def download_and_deserialize_uworker_output(output_url: str):
-  field.CopyFrom(serialized_json)
-
-
-class UworkerOutput(UworkerMsg):
-  """Class representing an unserialized UworkerOutput message from
-  utask_main."""
-  PROTO_CLS = uworker_msg_pb2.Output
-
-  def save_rich_type(self, attribute, value):
-    field = getattr(self.proto, attribute)
-    if isinstance(value, dict):
-      save_json_field(field, value)
-      return
-
-    # TODO(metzman): Remove this once everything is migrated. This is only
-    # needed because some functions need to support utasks and non-utasks at the
-    # same time.
-    if isinstance(value, uworker_msg_pb2.Input):
-      field.CopyFrom(value)
-      return
-
-    if isinstance(value, UworkerMsg):
-      field.CopyFrom(value.proto)
-      return
-
-    if not isinstance(value, UworkerEntityWrapper):
-      raise ValueError(f'{value} is of type {type(value)}. Can\'t serialize.')
-
-    wrapped_entity_proto = serialize_wrapped_entity(value)
-    field.CopyFrom(wrapped_entity_proto)
-
-
-class BuildData(UworkerMsg):
-  """Class representing an unserialized ProgressionTaskOutput message from
-  fuzz_task."""
-  PROTO_CLS = uworker_msg_pb2.BuildData
-
-  def save_rich_type(self, attribute, value):
-    field = getattr(self.proto, attribute)
-    if isinstance(value, (dict, list)):
-      save_json_field(field, value)
-      return
-
-    raise ValueError(f'{value} is of type {type(value)}. Can\'t serialize.')
-
-
-class UworkerInput(UworkerMsg):
-  """Class representing an unserialized UworkerInput message from
-  utask_preprocess."""
-  PROTO_CLS = uworker_msg_pb2.Input
-
-  def save_rich_type(self, attribute, value):
-    field = getattr(self.proto, attribute)
-    if isinstance(value, dict):
-      save_json_field(field, value)
-      return
-
-    if isinstance(field, collections.abc.Sequence):
-      # This the way to tell if it's a repeated field.
-      # We can't get the type of the repeated field directly.
-      value = list(value)
-      if len(value) == 0:
-        return
-      assert isinstance(value[0], ndb.Model), value[0]
-      field.extend([model._entity_to_protobuf(entity) for entity in value])  # pylint: disable=protected-access
-      return
-
-    if isinstance(value, UworkerMsg):
-      field.CopyFrom(value.proto)
-      return
-
-    if not isinstance(value, ndb.Model):
-      raise ValueError(f'{value} is of type {type(value)}. Can\'t serialize.')
+def model_to_protobuf(entity: ndb.Model) -> entity_pb2.Entity:
+  """Helper function to convert entity to protobuf format."""
+  return model._entity_to_protobuf(entity)  # pylint: disable=protected-access
 
-    entity_proto = model._entity_to_protobuf(value)  # pylint: disable=protected-access
-    field.CopyFrom(entity_proto)
 
+from typing import Type
+from typing import TypeVar
 
-class SetupInput(UworkerInput):
-  """Input for setup.update_fuzzer_and_data_bundle and setup.setup_testcase in
-  uworker_main."""
-  PROTO_CLS = uworker_msg_pb2.SetupInput
+T = TypeVar('T', bound=ndb.Model)
```

Comment:
I think this is overkill for now, because we do not have type-checking, so we can remove it. If you want to keep it, you'll want to move the imports to the top of the file.